src/ulid.cr

```diff
@@ -19,6 +20,49 @@ module ULID
     encode_time(seed_time, TIME_LEN) + encode_random(RANDOM_LEN)
   end
 
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid?("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => true
+  # ```
+  def valid?(ulid : String) : Bool
+    # Incorrect length && Invalid chars
+    ulid.size == TIME_LEN + RANDOM_LEN && ulid.upcase.chars.all? &.in?(ENCODING)
+  end
+
+  # Validate a string is a ULID
+  #
+  # ```
+  # ULID.valid!("01B3EAF48P97R8MP9WS6MHDTZ3")
+  # # => nil (or exception)
+  # ```
+  def valid!(ulid : String) : Nil
```

Comment:
Entirely optional, but what about `.validate!` for a little clearer API.

Reply:
nice

should I have
`.valid?` and `validate!`
or
`validate?` and `validate!`

Reply:
I'd go with option a - seems to flow nicely.